Remotion/Web/IntegrationTests/SmartPage/HistoryBackHandlingTests.cs

```diff
@@ -0,0 +1,159 @@
+      Assert.That(sessionStorageEntryKeyValue, Is.EqualTo((sessionStorageEntryCount - 1).ToString()));
+    }
+
+    [Test]
+    public void NavigateToDifferentPage_IncreasesSessionStorageEntries ()
+    {
+      var initialPage = StartWithCleanSessionStorage("WebButtonTest.wxe");
+      var jsExecutor = JavaScriptExecutor.GetJavaScriptExecutor(initialPage.Scope);
+      var initialSessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+
+      Start<WxePageObject>("WebTabStripTest.wxe");
+
+      var sessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+      var sessionStorageEntryKeyValue = GetSessionStorageEntry(jsExecutor, "pageTokenEntryCount");
+      var sessionStorageEntryCountDifference = sessionStorageEntryCount - initialSessionStorageEntryCount;
+      Assert.That(sessionStorageEntryCountDifference, Is.EqualTo(1));
+      Assert.That(sessionStorageEntryKeyValue, Is.EqualTo((sessionStorageEntryCount - 1).ToString()));
+    }
+
+    [Test]
+    public void NewSessionStorageEntry_OverThreshold_ClearsOldEntries ()
+    {
+      var initialPage = StartWithCleanSessionStorage("WebButtonTest.wxe");
+      var jsExecutor = JavaScriptExecutor.GetJavaScriptExecutor(initialPage.Scope);
+
+      foreach (var _ in Enumerable.Range(0, 149))
+        Start<WxePageObject>("WebTabStripTest.wxe");
+
+      Assert.That(GetSessionStorageEntryCount(jsExecutor), Is.EqualTo(151));
+      Assert.That(GetSessionStorageEntry(jsExecutor, "pageTokenEntryCount"), Is.EqualTo(150.ToString()));
+
+      Start<WxePageObject>("WebTabStripTest.wxe");
+
+      var sessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+      var sessionStorageEntryKeyValue = GetSessionStorageEntry(jsExecutor, "pageTokenEntryCount");
+      Assert.That(sessionStorageEntryCount, Is.EqualTo(101));
+      Assert.That(sessionStorageEntryKeyValue, Is.EqualTo((sessionStorageEntryCount - 1).ToString()));
+    }
+
+    [Test]
+    public void NavigatingBack_WithLastPageLoaded_ShowsOverlay ()
+    {
+      var home = StartWithCleanSessionStorage("WebButtonTest.wxe");
+      Start<WxePageObject>("WebTabStripTest.wxe");
+
+      home.Driver.GoBack(home.Scope);
+
+      _ = home.Scope.OuterHTML; // Ensure up to date html
+      var overlayScope = home.Scope.FindCss("#WxeStatusIsCachedMessage");
+      Assert.That(overlayScope.Exists(), Is.True);
+    }
+
+    [Test]
+    public void NavigatingBack_WithLastPageSubmitted_ShowsOverlay ()
+    {
+      var home = StartWithCleanSessionStorage("WebTabStripTest.wxe");
+      home.WebTabStrips().First().SwitchTo("Tab2");
+      home.WebTabStrips().First().SwitchTo("Tab1");
+
+      home.Driver.GoBack(home.Scope);
+
+      _ = home.Scope.OuterHTML; // Ensure up to date html
+      var overlayScope = home.Scope.FindCss("#WxeStatusIsCachedMessage");
+      Assert.That(overlayScope.Exists(), Is.True);
+    }
+
+    [Test]
+    public void NavigateForward_WithOverlayShown_IsNotPossible ()
```

Comment:
```suggestion
    public void NavigateForward_IsNotPossible ()
```

Unless I'm mistaken/blind, there's not test for the overlay or special operation, it's just load + back + forward, so we should use a more simple test name.